### PR TITLE
[artifactory][artifactory-ha] Add nginx customInitContainers, customVolumes, customSidecarContainers

### PR DESCRIPTION
**Pull request by kaychamo**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Chart Version bumped
- [ ] CHANGELOG.md updated
- [ ] Variables and other changes are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:
add support for custom elements in the nginx deployment

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
I/we need it for a nginx-fluentd sidecar

**Special notes for your reviewer**:
thanks in advance :)

### CODE REVIEW

**Comment by kaychamo** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by kaychamo** [comment]
@chukka may you please take a look at this PR and give me feedback before it can be merged

**Comment by chukka** [comment]
@kaychamo  Thanks for the contribution ! Can you explain us the use case here ?

File: stable/artifactory-ha/templates/_helpers.tpl [hunk 1]
```diff
@@ -407,3 +407,39 @@ artifactory port
 {{- .Values.router.internalPort  -}}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Resolve customInitContainers value
+*/}}
+{{- define "nginx.customInitContainers" -}}
```
Comment (chukka) [inline stable/artifactory-ha/templates/_helpers.tpl:414]: rename as `artifactory.nginx.customInitContainers`

**Comment by kaychamo** [comment]
thanks chukka :) 
so, we use this to add a nginx-sidecar-fluentd container in the nginx deployment
we also found that this project provide the nginx>fluentd.conf : https://github.com/jfrog/log-analytics-elastic
but the charts part wasn't yet covered

Thanks for your review and should i rename all 3 blocks in helpers or just this one mentionned?

**Comment by chukka** [comment]
> thanks chukka :) so, we use this to add a nginx-sidecar-fluentd container in the nginx deployment we also found that this project provide the nginx>fluentd.conf : https://github.com/jfrog/log-analytics-elastic but the charts part wasn't yet covered
> 
> Thanks for your review and should i rename all 3 blocks in helpers or just this one mentionned?

All three blocks. Please add same changes in artifactory-ha chart with `artifactory-ha.nginx.`

**Comment by rahulsadanandan** [comment]
@kaychamo thanks for the pr.
our implementation for `global` section was only meant for application related containers. 
Can we have this from only under nginx section not the global. 
cc @chukka

**Comment by kaychamo** [comment]
hiya @rahulsadanandan and thanks. you may check the last commit to see if it's OK now

**Comment by chukka** [comment]
Fixed in #1580

**Pull request closed without merging**